src/abstractitensornetwork.jl

```diff
@@ -935,6 +935,26 @@ function add(tn1::AbstractITensorNetwork, tn2::AbstractITensorNetwork)
   return tn12
 end
 
+""" Scale each tensor of the network by a scale factor on each vertex"""
+function scale!(tn::AbstractITensorNetwork, vertices_weights::Dictionary)
+  for v in keys(vertices_weights)
+    setindex_preserve_graph!(tn, vertices_weights[v] * tn[v], v)
+  end
+  return tn
+end
+
+""" Scale each tensor of the network via a function (vertex, ITensor) -> Number"""
+function scale!(tn::AbstractITensorNetwork, weight_function::Function)
```

Comment:
We should put the function first since that is conventional for Julia functions that accept other functions. Also, it would allow using the syntax:
```julia
function scale!(tn) do v
  # ...
end
```

Reply:
Also, it would make more sense to me for the function to just accept a vertex, not a vertex and a tensor (the tensor could be implicitly determined from the vertex if needed).